openssl-dynamic/src/main/c/sslutils.c

```diff
@@ -328,38 +326,53 @@ static unsigned char dhxxx2_g[]={
 
 static DH *get_dh(int idx)
 {
-    DH *dh;
-    if ((dh = DH_new()) == NULL)
+    DH *dh = NULL;
+    BIGNUM *p = NULL;
+    BIGNUM *g = NULL;
+    if ((dh = DH_new()) == NULL) {
         return NULL;
+    }
+
     switch (idx) {
         case SSL_TMP_KEY_DH_512:
-            dh->p = BN_bin2bn(dh0512_p, sizeof(dh0512_p), NULL);
+            p = BN_bin2bn(dh0512_p, sizeof(dh0512_p), NULL);
         break;
         case SSL_TMP_KEY_DH_1024:
-            dh->p = BN_bin2bn(dh1024_p, sizeof(dh1024_p), NULL);
+            p = BN_bin2bn(dh1024_p, sizeof(dh1024_p), NULL);
         break;
         case SSL_TMP_KEY_DH_2048:
-            dh->p = BN_bin2bn(dh2048_p, sizeof(dh2048_p), NULL);
+            p = BN_bin2bn(dh2048_p, sizeof(dh2048_p), NULL);
         break;
         case SSL_TMP_KEY_DH_4096:
-            dh->p = BN_bin2bn(dh4096_p, sizeof(dh2048_p), NULL);
+            p = BN_bin2bn(dh4096_p, sizeof(dh2048_p), NULL);
```

Comment:
@normanmaurer should be `sizeof(dh4096_p)` here

Reply:
yep... that said it should not matter :)

Reply:
Yep... good catch. This bug existed before btw.